### ISSUE TITLE
Use "ReferencedList" with a returned array

### ISSUE THREAD

**Issue by Xennis** [marmelab/ng-admin#241]
For instance a request to `http://localhost:3000/api/companies/65` returns

```
{
  "id": 65,
  "name": "Example Company",
  "employees": [
    {"id": 1, "name": "XY", "age": 25, ... },
    {"id": 3, "name": "YY", "age": 28, ... },
    ...
  ]
}
```

How can I use the returned `employees` array in combination with the field type `ReferencedList` (without making an extra API call, the returned array already contains all data)? Hence I want to use something like:
```javascript
var company = new Entity('companies');

company.showView().fields([
  new Field('name'),
  new ReferencedList('employees')
    //.targetEntity(employees)
    //.targetReferenceField('id')
    .targetFields([
       new Field('name'),
       new Field('age')
    ]);	
]);
```

Or if it doesn't work in combination with `ReferencedList`, how can I handle arrays like above instead?

**Comment by endel** [comment]
Our API can return nested related models too. It would be nice to have this option for this when calling `singleApiCall`, or something.

**Comment by docc** [comment]
Same here, heavily based on nested relations. It would be cool to be able to edit/add relations inline aswell.

**Comment by fzaninotto** [comment]
Great idea, this is something that `singleAPICall` should allow in the future.

**Comment by cvlmtg** [comment]
+1 for this feature.

**Comment by fzaninotto** [comment]
I believe this implies the creation of a new type, something like `embedded_list`. Until this is added to the core, you can still [write it yourself](https://github.com/marmelab/ng-admin/blob/master/doc/Custom-types.md), or use the `json` type.

**Comment by Phocea** [comment]
Merci Francois!
This is exactly what I require. Indeed the json type gives me a quick workaround for now.

Please bear in mind that this kind of nesting can even go deeper. Something like:

{
  "id": 65,
  "name": "Example Company",
  "employeeSet": [
    {"region": "France",
       "employees": [
           {"id": 1, "name": "XY", "age": 25, ... },
           {"id": 3, "name": "YY", "age": 28, ... }
       ]},
    {"region": "US",
       "employees": [
           {"id": 6, "name": "XY", "age": 29, ... },
           {"id": 8, "name": "YY", "age": 31, ... }
       ]}
    ]
}

**Comment by devguf** [comment]
+1

**Comment by SaiNageswarS** [comment]
+1

**Comment by AlexH7** [comment]
+1

**Comment by Phocea** [comment]
I have investigated this project https://github.com/jdorn/json-editor in order to go from a json to an html editor. Just thought I would post this here as it could be of great help toward a custom json / metadata editor

**Comment by fzaninotto** [comment]
Fixed by #714
